eZ/Bundle/EzPublishCoreBundle/Command/RemoveContentTranslationCommand.php

```diff
@@ -0,0 +1,214 @@
+use Symfony\Component\Console\Question\ChoiceQuestion;
+use Symfony\Component\Console\Question\ConfirmationQuestion;
+use Exception;
+
+/**
+ * Console Command which removes a given Translation from all the Versions of a given Content Object.
+ */
+class RemoveContentTranslationCommand extends Command
+{
+    /**
+     * @var \eZ\Publish\API\Repository\Repository
+     */
+    private $repository;
+
+    /**
+     * @var \eZ\Publish\API\Repository\ContentService
+     */
+    private $contentService;
+
+    /**
+     * @var \Symfony\Component\Console\Input\InputInterface
+     */
+    private $input;
+
+    /**
+     * @var \Symfony\Component\Console\Output\OutputInterface
+     */
+    private $output;
+
+    /**
+     * @var \Symfony\Component\Console\Helper\QuestionHelper
+     */
+    private $questionHelper;
+
+    public function __construct(Repository $repository)
+    {
+        parent::__construct(null);
+        $this->repository = $repository;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function configure()
+    {
+        $this
+            ->setName('ezplatform:remove_content_translation')
+            ->addArgument('content_id', InputArgument::REQUIRED, 'Content Object Id')
+            ->addArgument(
+                'lang_code',
+                InputArgument::REQUIRED,
+                'Language code of the Translation to be removed'
+            )
+            ->setDescription('Remove Translation from all the Versions of a Content Object');
+    }
+
+    protected function initialize(InputInterface $input, OutputInterface $output)
+    {
+        parent::initialize($input, $output);
+        $this->input = $input;
+        $this->output = $output;
+        $this->questionHelper = $this->getHelper('question');
+        $this->contentService = $this->repository->getContentService();
+
+        $this->repository->getPermissionResolver()->setCurrentUserReference(
+            $this->repository->getUserService()->loadUserByLogin('admin')
+        );
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function execute(InputInterface $input, OutputInterface $output)
+    {
+        $contentId = intval($input->getArgument('content_id'));
+        $languageCode = $input->getArgument('lang_code');
+
+        if ($contentId === 0) {
+            throw new InvalidArgumentException(
+                'content_id',
+                'Content Object Id has to be an integer'
+            );
+        }
+
+        $this->output->writeln(
+            '<comment>**NOTE**: Make sure to run this command using the same SYMFONY_ENV setting as your eZ Platform installation does</comment>'
+        );
+
+        $contentInfo = $this->contentService->loadContentInfo($contentId);
+
+        $this->repository->beginTransaction();
+        try {
+            // Cleanup Content before removing Translation
+            $contentInfo = $this->cleanupContent($contentInfo, $languageCode);
```

Comment:
can make the code more readable if it was
```php
$allLanguages = $this->removeAffectedSingularLanguageVersions($contentInfo, $languageCode);
if ($contentInfo->mainLanguageCode === $languageCode) {
    $contentInfo = $this->promtUserForMainLanuguageChange($contentInfo, $languageCode, $allLanguages);
}
```

Reply:
Ok, squashed into 8f8412b